projects/output-and-error-handling/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+)
+
+func sendRequest(url string) *http.Response {
+	resp, err := http.Get(url)
+	if err != nil {
+		fmt.Println("We can't get the weather!")
+		fmt.Fprintln(os.Stderr, "Failed to initialize.!", err)
+		os.Exit(2)
+	}
+	if resp == nil {
+		fmt.Fprintln(os.Stderr, "The server did not reply anything, which here is considered an error!")
+		os.Exit(52)
+	}
+	return resp
+}
+
+func handleStatusCode(resp *http.Response, sleepTime *time.Duration) {
+	defer resp.Body.Close()
+	if resp.StatusCode == 200 {
+		scanner := bufio.NewScanner(resp.Body)
+		for i := 0; scanner.Scan(); i++ {
+			fmt.Println(scanner.Text())
+		}
+		if err := scanner.Err(); err != nil {
+			panic(err)
+		}
+		os.Exit(0)
```

Comment:
As above, we tend to avoid panicking or exiting in functions - I'd have this function return an `error`, and have that be `nil` in the "everything was fine" case, and deal with the exiting in `main` itself